File: 2681/Solution.java
import java.util.Arrays;

class Solution {
    public int sumOfPower(int[] nums) {
        int n = nums.length;
        int mod = 1000000007;
        int res = 0, dp = 0, preSum = 0;
        Arrays.sort(nums);
        for (int i = 0; i < n; i++) {
            dp = (nums[i] + preSum) % mod;
            preSum = (preSum + dp) % mod;
            res = (int) ((res + (long) nums[i] * nums[i] % mod * dp) % mod);
            if (res < 0) {
                res += mod;
            }
        }
        return res;
    }
}

File: 2681/solution.go
package _2681

import "sort"

func sumOfPower(nums []int) int {
	mod := int(1e9 + 7)
	n := len(nums)
	sort.Ints(nums)
	res := 0
	dp := 0
	preSum := 0
	for i := 0; i < n; i++ {
		dp = (preSum + nums[i]) % mod
		preSum = (preSum + dp) % mod
		res = (res + (dp*nums[i]%mod)*nums[i]%mod) % mod
	}
	return res
}
